dev/local-environment/configurations/pc-contracts-cli/entrypoint.sh

```diff
@@ -0,0 +1,346 @@
+PC_CONTRACTS_CLI_ZIP_URL="https://github.com/input-output-hk/partner-chains-smart-contracts/releases/download/v6.1.0/trustless-sidechain-cli-6.1.0-x86_64-linux.zip"
+PARTNER_CHAINS_NODE_URL="https://github.com/input-output-hk/partner-chains/releases/download/1.1.1-rc1/partner-chains-node-1.1.1-rc1-x86_64-linux"
+PARTNER_CHAINS_CLI_URL="https://github.com/input-output-hk/partner-chains/releases/download/1.1.1-rc1/partner-chains-cli-1.1.1-rc1-x86_64-linux"
+
+# Initialize flags
+PC_NODE_READY=0
+PC_CLI_READY=0
+PC_CONTRACTS_CLI_READY=0
+
+if [ "$ARTIFACT_OVERRIDE" == "yes" ]; then
+  echo "Artifact override is enabled. Checking for local artifacts..."
+
+  # Check and set flags for existing artifacts, and copy if found
+  if [ -f "/overrides/pc-contracts-cli" ] && [ -d "/overrides/node_modules" ]; then
+    echo "pc-contracts-cli and node_modules found in /overrides/. Using local artifacts."
+    cp /overrides/pc-contracts-cli ./pc-contracts-cli
+    cp -r /overrides/node_modules ./node_modules
+    echo "pc-contracts-cli and node_modules copied."
+    PC_CONTRACTS_CLI_READY=1
+  fi
+
+  if [ -f "/overrides/partner-chains-node" ]; then
+    echo "partner-chains-node found in /overrides/. Using local artifact."
+    cp /overrides/partner-chains-node ./partner-chains-node
+    echo "partner-chains-node copied."
+    PC_NODE_READY=1
+  fi
+
+  if [ -f "/overrides/partner-chains-cli" ]; then
+    echo "partner-chains-cli found in /overrides/. Using local artifact."
+    cp /overrides/partner-chains-cli ./partner-chains-cli
+    echo "partner-chains-cli copied."
+    PC_CLI_READY=1
+  fi
+
+else
+  echo "Artifact override is not enabled. Defaulting to downloading all artifacts..."
+fi
+
+# Check which artifacts need to be downloaded
+if [ "$PC_CONTRACTS_CLI_READY" -eq 0 ]; then
+  echo "Downloading pc-contracts-cli and node_modules..."
+  wget -q -O ./pc-contracts-cli.zip "$PC_CONTRACTS_CLI_ZIP_URL"
+  unzip -o ./pc-contracts-cli.zip > /dev/null
+  mv ./sidechain-cli ./pc-contracts-cli
+fi
+
+if [ "$PC_NODE_READY" -eq 0 ]; then
+  echo "Downloading partner-chains-node..."
+  wget -q -O ./partner-chains-node "$PARTNER_CHAINS_NODE_URL"
+fi
+
+if [ "$PC_CLI_READY" -eq 0 ]; then
+  echo "Downloading partner-chains-cli..."
+  wget -q -O ./partner-chains-cli "$PARTNER_CHAINS_CLI_URL"
+fi
+
+# Set executable permissions
+chmod +x ./partner-chains-node
+chmod +x ./partner-chains-cli
+chmod +x ./pc-contracts-cli
+
+# Install jq
+apt -qq update &> /dev/null
+apt -qq -y install jq netcat &> /dev/null
+
+echo "Dependencies downloaded and binaries made executable."
+
+# Wait for the node to start
+echo "Waiting for the Cardano network to sync and for Kupo and Ogmios to start..."
+
+while true; do
+    if nc -z kupo $KUPO_PORT && nc -z ogmios $OGMIOS_PORT; then
+        break
+    else
+        sleep 10
+    fi
+done
+
+echo "Kupo and Ogmios have started. Waiting 10 seconds for them to stabilise, and then will begin configuration..."
```

Comment:
Ogmios is stateless, so it doesn't need time to sync. Kupo promises a header that tells you slot of the last block it knows, but I have not seen it in the output, so perhaps we have to settle on this.
Writing this comment to this env developers to be aware.